### PR TITLE
chore: add module wide babel & browserify configs to fix 3rd party builds

### LINKED ISSUE #412: It broke for browserify
Browserify does not check the existence of the "browser-field" in the npm package.json. Even with babelify, it does not work.

### PR DESCRIPTION
**Pull request by thgreasi**
Properly resolves #412 & #413 for the ES6 codebase.

### CODE REVIEW

**Comment by thgreasi** [comment]
The whole point of this PR it to fix localforage for browserify users, before changing the current build system so that we point all users to the `dist` files.

It still points browserify to the `src` files (like before the babel&es6 changes) but discloses the transformations and configurations required to build localforage sources with `browserify` (specifically `babelify` and `system-import-transformer`).

In my tests [open-flight-planner](https://github.com/Turbo87/open-flight-planner) seems to work, requiring only `system-import-transformer`.
@rejas, @Turbo87, @bhamblok can you cross check that your apps work with [this branch](https://github.com/thgreasi/localForage/tree/3rdparty-babelify-build-fix)?

**Comment by rejas** [comment]
thx, will take a look at it tomorrow morning

**Comment by rejas** [comment]
Not sure, maybe I am doing somehting wrong, but I get when using your branch:

[11:03:17] Browserify Error { [ReferenceError: Unknown plugin "system-import-transformer" while parsing file: /Users/mveeck/Projekte/.../node_modules/ampersand-sync-localforage/node_modules/localforage/src/localforage.js]

**Comment by thgreasi** [comment]
**TL;DR** You need to install [system-import-transformer](https://github.com/thgreasi/babel-plugin-system-import-transformer).

This is absolutely expected, since when using browserify to build a `node_module` into your dist, you need to have all the required transformers and plugins installed.
Browserify would also complain in case you were missing babelily itself, since it is [a dependency specified for the browserify build in package.json](https://github.com/thgreasi/localForage/blob/8b34f2904f90ede9267244e685bf39d5d7acfead/package.json#L52-L54).

As I said earlier, this is just a quick fix to allow browserify users to use the code in the master branch again. This is obviously not an ideal solution but just a temporary one. The next step is to point them to a browserify-friendly dist file (unfortunately this was never the case before).

**Comment by rejas** [comment]
Installed it, and it worked fine. :thumbsup:  Thanks for the work, hope it gets merged.

**Comment by thgreasi** [comment]
Just pushed 352d258a44c7996b659a90d3bab672c5b0fe18b4 that modifies the browserify build test, to use the module sources (and build options) as defined in package.json.
Hopefully this will prevent us from breaking localforage for browserify users again.

**Comment by tofumatt** [comment]
Sweeeeet!

This means the test is using the same build that anyone else using browserify would use now, right? :+1:

**Comment by thgreasi** [comment]
Yea, more or less.
Most of the browserify build options have moved to package.json and a babel configuration file. Even in the test case we require the main repo directory, making browserify read package.json for the build options and file paths.
Next thing is to do the same for webpack testing.
